includes/payment-methods/class-wc-gateway-stripe-boleto.php

```diff
@@ -53,6 +53,66 @@ public function __construct() {
 		parent::__construct();
 
 		add_filter( 'wc_stripe_allowed_payment_processing_statuses', [ $this, 'add_allowed_payment_processing_statuses' ], 10, 2 );
+		add_filter( 'wc_stripe_payment_gateway_settings', [ $this, 'add_expiration_option' ], 10, 2 );
+		add_action( 'wc_stripe_update_payment_gateway_settings', [ $this, 'update_expiration_option' ], 10, 2 );
+		add_filter( 'woocommerce_stripe_request_body', [ $this, 'add_expires_after_days' ], 10, 2 );
```

Comment:
In the same line of thought as my comment [above](https://github.com/woocommerce/woocommerce-gateway-stripe/pull/2412/files#r952940907) about using hooks, we could instead have a protected function for Boleto's gateway class to modify the request body [directly when we're creating and updating](https://github.com/woocommerce/woocommerce-gateway-stripe/blob/develop/includes/abstracts/abstract-wc-stripe-payment-gateway-voucher.php#L309) the intent.

Reply:
`woocommerce_stripe_request_body` already exists [here](https://github.com/woocommerce/woocommerce-gateway-stripe/blob/17c45385eecf8e11752a5544b13d0ba20a373b5d/includes/class-wc-stripe-api.php#L133).  We could use a protected function, but I think it complicates the gateway without added protection.  What's the benefit of not using the existing filter here?

Reply:
The main benefit I see of directly updating the intent's properties instead of using a hook is readability. If I were trying to see where the properties for Boleto's payment intents, I'd expect to see all the related logic in a somewhat related space. In this case, the `create_or_update_payment_intent` method. I find it tricky to notice that we're hooking into the `woocommerce_stripe_request_body` filter for adding extra parameters to the intent.

Another detail that I consider a disadvantage of using the hook for this case is that we add an extra thing to take care of for external developers using the `woocommerce_stripe_request_body` hook, since they'd need to keep in mind the priority of our filter when they're hooking into it. But the readability point is what I'm most concerned about.

Glad to have a different perspective here, though! We could bring this week's PR reviewer or anyone else if that's good with you.